core/src/main/java/com/novoda/noplayer/PlayerView.java

```diff
@@ -19,4 +21,57 @@
     void hideSubtitles();
 
     void setSubtitleCue(TextCues textCues);
+
+    class SurfaceContainer {
+
+        @Nullable
+        private final SurfaceView surfaceView;
+        @Nullable
+        private final TextureView textureView;
+        private final PlayerViewSurfaceHolder surfaceHolder;
+
+        public static SurfaceContainer create(SurfaceView surfaceView) {
+            PlayerViewSurfaceHolder surfaceHolder = new PlayerViewSurfaceHolder();
+            surfaceView.getHolder().addCallback(surfaceHolder);
+            return new SurfaceContainer(surfaceView, null, surfaceHolder);
+        }
+
+        public static SurfaceContainer create(TextureView textureView) {
+            PlayerViewSurfaceHolder surfaceHolder = new PlayerViewSurfaceHolder();
+            textureView.setSurfaceTextureListener(surfaceHolder);
+            return new SurfaceContainer(null, textureView, new PlayerViewSurfaceHolder());
+        }
+
+        SurfaceContainer(@Nullable SurfaceView surfaceView, @Nullable TextureView textureView, PlayerViewSurfaceHolder surfaceHolder) {
```

Comment:
Shouldn't this be private?

Reply:
then the PMD and checkstyle complain that the class should be `final` and then I cannot mock it in tests

Reply:
You can mock final classes in tests, there is a hack to do it, I thought we had it in `no-player` 🤔

Nope, turns out we have it in `download-manager`
 https://github.com/novoda/download-manager/blob/epic_download_manager_v2/library/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker

https://github.com/mockito/mockito/wiki/What's-new-in-Mockito-2#mock-the-unmockable-opt-in-mocking-of-final-classesmethods